### DISCUSSION TITLE
Compiler conflicts between packages

### DISCUSSION THREAD

**Discussion by daviddcrrl** [dotnet/maui#16495] [Upgrading from Xamarin]
Hi, I'm upgrading a project from Xamarin.Forms to Maui, and I'm facing a problem I cannot resolve. If anyone could help me I'd really appreciate that.

I'm getting error XA4215, message "The Java type 'mono.android.support.v4.media.session.MediaSessionCompat_OnActiveChangeListenerImplementor' is generated by more than one managed type. Please change the [Register] attribute so that the same Java type is not emitted."
By looking at other related topics, I've assumed there is conflict by two NuGet packages or dependencies.

My project has as dependencies: CommunityToolkit.Maui, FreshMvvm.Maui, Geolocator.Plugin, Microsoft.Extensions.Login.Debug, Microsoft.TeamFoundationServer.ExtendedClient, Newtonsoft.Json, NuGet.Configuration, Plugin.Permissions, PropertyChanged.Fody, and sqlite-net-pcl.

I'd really appreciate some tips and advice. If you need more info to help me just let me know.

**Comment by nll** [comment]
This error points to an incompatible packages (normally from Xamarin.Android group) - e.g we refer package A and package B but some intermediate dependencies are incompatible, like package X published 2 year ago and Y publish 2 months ago. The error message will also tell us about the two jar locations that are causing the problem so we can find out what packages are causing the problem.

In this particular case it seems that you're using an old Android support package - `Xamarin.Android.Support.v4` - that is no longer supported in MAUI, and it's probably in conflict with its AndroidX version. I would suggest to first try to delete all `obj` and `bin` folder as it could be just local stale files. Then I would try to discover what is requiring that particular dependency - for instance looking at `obj/project.assets.json` and searching for "Xamarin.Android.Support.v4" - and try to remove or update the culprit.
